### PR TITLE
Mrtk 2017.2 forward port

### LINKED ISSUE #851: HTK Build Window not compatible with Immersive Headset build and deploy
Right now the Build Window is specifically targeted at building out to either the emulator or to IpOverUsb and deploying to device.  It is not capable of deploying to local machine like is needed for the immersive headsets.

### LINKED ISSUE #895: We should reconsider the Set Scene Settings option(s) that are in the Toolkit Menu
Those settings right now are very HoloLens centric.  Almost all the settings that are being done in that are not what is needed for Immersive Headsets.  I would like to see a discussion on how this can be better implemented for all cases, Immersive Apps, Holographic Apps, and Hybrid/Asymmetric Apps.

### LINKED ISSUE #1018: Update Build Window with options for new HMD targets
Add more options for specifying a target device.

### LINKED ISSUE #1019: Update Mixed Reality Project Settings Wizard
Update the Mixed Reality Project Settings wizard with device specific capabilities.

### PR DESCRIPTION
**Pull request by StephenHodgson**
Patch of #1048 
Fixes #851
Fixes #895
Fixes #1018
Fixes #1019

### CODE REVIEW

File: Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
     /// <summary>
-    /// Renders the UI and handles update logic for HoloToolkit/Configure/Apply Mixed Reality Scene Settings.
+    /// Renders the UI and handles update logic for MixedRealityToolkit/Configure/Apply Mixed Reality Scene Settings.
     /// </summary>
     public class SceneSettingsWindow : AutoConfigureWindow<SceneSettingsWindow.SceneSetting>
     {
+        /// <summary>
+        /// Can be found in the meta file of the camera prefab.  We use the GUID instead of a path in case people move the toolkit folders & assets around in their own projects.
+        /// </summary>
+        private const string CameraPrefabGUID = "f46f9eddea7bd4d4e813283cdc60cb96";
```
Comment (keveleigh) [inline Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: This is the HoloLensCamera prefab GUID. What about the other camera prefabs?

We should probably go down to 1 prefab anyway, which adapts to any situation.

Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: Pretty sure I updated the GUID to point to be the MixedRealityCamera prefab.

Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: @keveleigh would you prefer the `MiedRealityCamera` prefab, or the `MixedRealityCameraParent` prefab?

Comment (keveleigh) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: Weird, it was definitely a different GUID when I commented, though my window must have been out of date!

If we want to go down to 1 prefab, reusing the HoloLensCamera GUID would probably be best to maintain forward compatibility, right?

Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: >If we want to go down to 1 prefab, reusing the HoloLensCamera GUID would probably be best to maintain forward compatibility, right?

I don't think so.  I'd fear the new branch would get a lot of missing references if we changed that.

I think it's fine the way it is anyway, cause the scene settings window will just replace the camera in the scene anyway.  That way devs have the option to update, or use the new prefab.

Comment (keveleigh) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: Cool, in that case, MixedRealityCameraParent would be the preferred one to use.

Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: I'll update the GUID to point to it.

**Comment by StephenHodgson** [comment]
See #1059

**Comment by keveleigh** [comment]
Are these port PRs different from the ones going into master? Could we merge the master PRs then do another merge from master to the dev branch, so we don't end up with duplicate commits?

**Comment by StephenHodgson** [comment]
@keveleigh I created patches, so the commits shouldn't have conflicts.

**Comment by StephenHodgson** [comment]
Although my latest PR might have broken that cause I had to redo it because I messed up the rebase.

**Comment by StephenHodgson** [comment]
I'll close this in favor of the master branch PR and do master merge syncs instead.

**Pull request closed without merging**